packages/core/components/Puck/components/Header/index.tsx

```diff
@@ -79,7 +79,7 @@ const HeaderInner = <
     }
 
     return DefaultOverride;
-  }, [renderHeader]);
+  }, [renderHeader, renderHeaderActions]);
```

Comment:
The other dependency can likely be removed

```suggestion
  }, [renderHeaderActions]);
```